### DISCUSSION TITLE
Setting to Not Delete Git History of User Changes When Rebase Occurs

### DISCUSSION THREAD

**Discussion by anderson-j-molter** [renovatebot/renovate#27525] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

github.com

### What is your question?

We have a renovate bot that updates our node.js project. Sometimes version updates require changes manually by the user due to the version fix. However, if the user commits a change to the branch that the renovate bot opened, and then a rebase is run after the fact. The git history is wiped from the branch entirely. I understand that the bot wants to rebase from main and avoid conflicts so it goes to the most recent version of main and then applies the version update, but is there any way to configure it to not remove the previous git history? That way if this occurs the user can at least see the necessary changes to take in the commit history and apply them again.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by viceice** [comment]
renovate shouldn't tease modified branches.
don't amend changes.
if you explicitly request a rebase, then renovate will not rebase, it will create and force-push a complete new commit.
this is by design.

**Comment by rarkins** [comment]
Renovate won't force push any branch which has been modified by another account. If you are modifying renovated branches and setting the git author to the same as Renovate, stop doing that and add git commits the usual way so that renovate can detect the modification.
